### ISSUE TITLE
[Module Proposal]: `avm-res-web-connection`

### ISSUE THREAD

**Issue by PmeshramPM** [Azure/Azure-Verified-Modules#1105]
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes.

### Bicep or Terraform?

Terraform

### Module Classification?

Resource Module

### Module Name

avm-res-web-connection

### Module Details

AVM Resource Module for API Connection

### Do you want to be the owner of this module?

No

**Comment by dukumili** [comment]
I Confirm that I wish to own this AVM module and want to understand all the requirements and definition of a module owner. Thank You!

**Comment by prjelesi** [comment]
HI @dukumili 

Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!

Before starting development, please ensure ALL the following requirements are met.

**Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:

- For your module:
  - `ModuleName` - for naming your module
  - `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
- For your module's repository:
  - Repo name and folder path are defined in `RepoURL`
  - Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
  - Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).

Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍

You can now start the development of this module! ✅ Happy coding! 🎉

**Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**

Any further questions or clarifications needed, let us know!

Thanks,

The AVM Core Team

**Comment by dukumili** [comment]
> HI @dukumili
> 
> Thanks for confirming that you wish to own this AVM module and understand the related requirements and responsibilities!
> 
> Before starting development, please ensure ALL the following requirements are met.
> 
> **Please use the following values explicitly as provided in the [module index](https://azure.github.io/Azure-Verified-Modules/indexes/) page**:
> 
> * For your module:
>   
>   * `ModuleName` - for naming your module
>   * `TelemetryIdPrefix` - for your module's [telemetry](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr3---category-telemetry---deploymentusage-telemetry)
> * For your module's repository:
>   
>   * Repo name and folder path are defined in `RepoURL`
>   * Create GitHub teams for module owners and contributors and grant them permissions as outlined [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).
>   * Grant permissions for the AVM core team and PG teams on your GitHub repo as described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr9---category-contributionsupport---avm--pg-teams-github-repo-permissions).
> 
> Check if this module exists in the other IaC language. If so, collaborate with the other owner for consistency. 👍
> 
> You can now start the development of this module! ✅ Happy coding! 🎉
> 
> **Please respond to this comment and request a review from the AVM core team once your module is ready to be published! Please include a link pointing to your PR, once available. 🙏**
> 
> Any further questions or clarifications needed, let us know!
> 
> Thanks,
> 
> The AVM Core Team

@prjelesi I have updated the Terraform module for API Connection, please verify my PR and approve accordingly <https://github.com/Azure/terraform-azurerm-avm-res-web-connection/pull/3>

**Comment by PmeshramPM** [comment]
Moving this back to proposed status since the assignee is not longer working for MS.

**Comment by donovm4** [comment]
@jaredfholgate + @PmeshramPM : Is this a module that I can pick up?

**Comment by matebarabas** [comment]
Setting status to "needs module owner", as the previous owner no longer works at Microsoft.

**Comment by matebarabas** [comment]
@donovm4, on behalf of the team, apologies for the delayed response. Are you still interested in picking this one up? Thanks!

CC: @jeanchg

**Comment by donovm4** [comment]
I am indeed still interested in picking this module up!

cc: @jeanchg + @matebarabas

**Comment by jeanchg** [comment]
Thanks a lot @donovm4 ! 
I'm updating the index and repo's ownership accordingly.
@jaredfholgate could you pls add @donovm4 to both contributors and maintainers teams for the repo? Thank you

**Comment by donovm4** [comment]
Hi all, I have finished the initial development of this module. Any next steps before publishing to registry?
cc: @jeanchg, @jaredfholgate, @matebarabas, @prjelesi

**Comment by jeanchg** [comment]
hey @donovm4 , pls go ahead and publish your module (it's not your first module, so you know the process but here is the guidance if needed: https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/owner-contribution-flow/#5-publish-the-module).
Pls share the PR here when merged to main and let me know when published so I can update the index and we'll also close this issue.

**Comment by donovm4** [comment]
Hi @jeanchg : This is the [PR](https://github.com/Azure/terraform-azurerm-avm-res-web-connection/pull/17) that was already developed, tested, and reviewed.

I have created a draft but there seems to be some blocks. I can only JIT access as Administrator for only a few minutes, before it gets revoked. Also, are we no longer able to sign into the registry with GitHub? The HashiCorp registry prompts me for an HCP Terraform login, instead of the GitHub login as previously used.

**Comment by jaredfholgate** [comment]
> Hi [@jeanchg](https://github.com/jeanchg) : This is the [PR](https://github.com/Azure/terraform-azurerm-avm-res-web-connection/pull/17) that was already developed, tested, and reviewed.
> 
> I have created a draft but there seems to be some blocks. I can only JIT access as Administrator for only a few minutes, before it gets revoked. Also, are we no longer able to sign into the registry with GitHub? The HashiCorp registry prompts me for an HCP Terraform login, instead of the GitHub login as previously used.

Hi @donovm4. Click this link here or go directly to https://registry.terraform.io/sign-in/legacy to login with GitHub:

<img width="1127" height="558" alt="Image" src="https://github.com/user-attachments/assets/dd86f265-e7f3-4235-a28c-00ec4517ce43" />

**Comment by jeanchg** [comment]
@donovm4 anything you need to publish the first version, pls let us know

**Comment by donovm4** [comment]
@jaredfholgate + @jeanchg : do I need to elevate permissions before being able to successfully publish? I made it in through GitHub login and can select the repo, but when I try to publish nothing returns on the page.

**Comment by jaredfholgate** [comment]
Yes, try using JIT to elevate as it needs permission to add a webhook to your repo.

**Comment by donovm4** [comment]
Module has been published [here](https://registry.terraform.io/modules/Azure/avm-res-web-connection/azurerm/latest) as version `v0.1.0`.

cc: @jaredfholgate, @jeanchg, @matebarabas, @prjelesi

**Comment by jeanchg** [comment]
thank you @donovm4 !! I'm updating the index and closing this issue
